src/modules/hip/kernel/phase.hpp

```diff
@@ -0,0 +1,276 @@
+    dst_f8->f1[1] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[1] + 128, src2_f8->f1[1] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[2] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[2] + 128, src2_f8->f1[2] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[3] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[3] + 128, src2_f8->f1[3] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[4] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[4] + 128, src2_f8->f1[4] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[5] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[5] + 128, src2_f8->f1[5] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[6] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[6] + 128, src2_f8->f1[6] + 128) * *multiplier_f1)) - 128;
+    dst_f8->f1[7] = rpp_hip_pixel_check_0to255(truncf(atan2f(src1_f8->f1[7] + 128, src2_f8->f1[7] + 128) * *multiplier_f1)) - 128;
+}
+
+__device__ void phase_hip_compute(half *srcPtr, d_float8 *src1_f8, d_float8 *src2_f8, d_float8 *dst_f8, float *multiplier_f1)
+{
+    dst_f8->f1[0] = atan2f(src1_f8->f1[0], src2_f8->f1[0]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[1] = atan2f(src1_f8->f1[1], src2_f8->f1[1]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[2] = atan2f(src1_f8->f1[2], src2_f8->f1[2]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[3] = atan2f(src1_f8->f1[3], src2_f8->f1[3]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[4] = atan2f(src1_f8->f1[4], src2_f8->f1[4]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[5] = atan2f(src1_f8->f1[5], src2_f8->f1[5]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[6] = atan2f(src1_f8->f1[6], src2_f8->f1[6]) * *multiplier_f1 * ONE_OVER_255;
+    dst_f8->f1[7] = atan2f(src1_f8->f1[7], src2_f8->f1[7]) * *multiplier_f1 * ONE_OVER_255;
+}
+
+template <typename T>
+__global__ void phase_pkd_hip_tensor(T *srcPtr1,
+                                     T *srcPtr2,
+                                     uint2 srcStridesNH,
+                                     T *dstPtr,
+                                     uint2 dstStridesNH,
+                                     RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth * 3))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x * 3);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x;
+
+    float multiplier_f1 = (float)(255 / 1.570796);
+
+    d_float8 src1_f8, src2_f8, dst_f8;
+
+    rpp_hip_load8_and_unpack_to_float8(srcPtr1 + srcIdx, &src1_f8);
```

Comment:
can you try with load 24 and store 24 like how it is being done in latest kernels

Reference as below 
https://github.com/snehaa8/rpp/blob/sn/phase_tensor/src/modules/hip/kernel/color_twist.hpp#L148

Please check the performance and retain the best version

Reply:
Yes seeing slight improvement at times with load and store of 24 pixels